src/Core/Content/Product/Cart/ProductCartProcessor.php

```diff
@@ -102,6 +102,8 @@ public function collect(CartDataCollection $data, Cart $original, SalesChannelCo
                     if (\in_array($lineItem->getReferencedId(), $products->getIds(), true)) {
                         $lineItem->setDataTimestamp(new \DateTimeImmutable());
                         $lineItem->setDataContextHash($hash);
+                    } else {
+                        $lineItem->setDataTimestamp(null);
```

Comment:
Can you add an test case for this?

Reply:
Already on it

Testing Shopware\Core\Content\Test\Product\Cart\ProductCartProcessorTest
................................                                  33 / 33 (100%)